keyboards/mechwild/sugarglider/sugarglider.c

```diff
@@ -0,0 +1,438 @@
+#define DPI_OPTION_SIZE (sizeof(dpi_array) / sizeof(uint16_t))
+
+void board_init(void) {
+    // B9 is configured as I2C1_SDA in the board file; that function must be
+    // disabled before using B7 as I2C1_SDA.
+    setPinInputHigh(B9);
+    setPinOutput(B12);
+    setPinOutput(B13);
+    setPinOutput(B14);
+    writePinLow(B12);
+    writePinLow(B13);
+    writePinLow(B14);
+    setPinOutput(C13);
+}
+
+#ifdef DYNAMIC_TAPPING_TERM_ENABLE
+void tap_modify(int change_value, bool tap_status) {
+    if (keyboard_config.dt_term_config < 0) {
+        keyboard_config.dt_term_config *= -1;
+    }
+
+    keyboard_config.dt_term_config += change_value;
+
+    if (tap_status == false ) {
+        keyboard_config.dt_term_config *= -1;
+        g_tapping_term = 0;
+    } else {
+        g_tapping_term = keyboard_config.dt_term_config;
+    }
+    eeconfig_update_kb(keyboard_config.raw);
+}
+
+void tap_toggle(void) {
+    keyboard_config.dt_term_config *= -1;
+    if (keyboard_config.dt_term_config > 0) {
+        g_tapping_term = keyboard_config.dt_term_config;
+    } else {
+        g_tapping_term = 0;
+    }
+    eeconfig_update_kb(keyboard_config.raw);
+}
+#endif
+
+#ifdef DIP_SWITCH_ENABLE
+bool dip_switch_update_kb(uint8_t index, bool active) {
+    if (!dip_switch_update_user(index, active)) { return false; }
+    switch (index) {
+        case 0:
+            if(active != host_keyboard_led_state().caps_lock) { tap_code(KC_CAPS); }
+            break;
+        break;
+    }
+    return true;
+}
+#endif
+
+#if defined(ENCODER_ENABLE) && ! defined(ENCODER_MAP_ENABLE)
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+    if (!encoder_update_user(index, clockwise)) { return false; }
+    switch (index) {
+        case 0:
+            if (clockwise) {
+                tap_code(KC_VOLU);
+            } else {
+                tap_code(KC_VOLD);
+            }
+        break;
+        case 1:
+            if (clockwise) {
+                tap_code(KC_VOLU);
+            } else {
+                tap_code(KC_VOLD);
+            }
+        break;
+        case 2:
+            if (clockwise) {
+                tap_code(KC_PGUP);
+            } else {
+                tap_code(KC_PGDN);
+            }
+        break;
+        case 3:
+            if (clockwise) {
+                tap_code(KC_WH_U);
+            } else {
+                tap_code(KC_WH_D);
+            }
+        break;
+    }
+    return true;
+}
+#endif
+
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if(res) {
+        writePin(B12, led_state.num_lock);     // Updates status LEDs
+        writePin(B13, led_state.caps_lock);    // Updates status LEDs
+        writePin(B14, led_state.scroll_lock);  // Updates status LEDs
+        writePin(C13, !led_state.caps_lock);   // Updates status LEDs, this is the LED on the blackpill itself
+    }
+    return res;
+}
+
+#if defined(OLED_ENABLE) && defined(OLED_DISPLAY_128X64) // Wide OLED Functionality
```

Comment:
I mostly agree with this. This is another artifact of the original code since I haven't bothered with a full rewrite. I will consider this change while looking at the other changes that you have suggested.